x/auth/ante/unordered.go

```diff
@@ -62,7 +78,39 @@ func (d *UnorderedTxDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, _ bool, ne
 		return ctx, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "unordered tx ttl exceeds %d", d.maxUnOrderedTTL)
 	}
 
-	txHash := sha256.Sum256(ctx.TxBytes())
+	// in order to create a deterministic hash based on the tx, we need to hash the contents of the tx with signature
+	// Get a Buffer from the pool
+	buf := bufPool.Get().(*bytes.Buffer)
+	// Make sure to reset the buffer
+	buf.Reset()
+
+	// Use the buffer
+	for _, msg := range tx.GetMsgs() {
+		// loop through the messages and write them to the buffer
+		// encoding the msg to bytes makes it deterministic within the state machine.
+		// Malleability is not a concern here because the state machine will encode the transaction deterministically.
+		bz, err := proto.Marshal(msg)
```

Comment:
if we could get the "google.Protobuf.Any" version we would avoid the extra marshal

Reply:
i don't think so since you already have the bytes, cc @aaronc

Reply:
Malleability could be an issue because then you would have the client's encoding which could differ from the node's encoding and also what was signed over in the case of amino

Reply:
this is my understanding of it as well. unordered tx has the trade off of an extra marshal